integration/tests/bft/base_suite.go

```diff
@@ -0,0 +1,152 @@
+package bft
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/onflow/flow-go/engine/ghost/client"
+	"github.com/onflow/flow-go/insecure"
+	"github.com/onflow/flow-go/insecure/orchestrator"
+	"github.com/onflow/flow-go/integration/testnet"
+	"github.com/onflow/flow-go/integration/tests/lib"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+type BaseSuite struct {
+	suite.Suite
+	Log                     zerolog.Logger
+	lib.TestnetStateTracker // used to track messages over testnet
+	Ctx                     context.Context
+	Cancel                  context.CancelFunc   // used to tear down the testnet
+	Net                     *testnet.FlowNetwork // used to keep an instance of testnet
+	GhostID                 flow.Identifier      // represents id of ghost node
+	NodeConfigs             testnet.NodeConfigs  // used to keep configuration of nodes in testnet
+	OrchestratorNetwork     *orchestrator.Network
+	BlockRateFlag           string
+}
+
+// Ghost returns a client to interact with the Ghost node on testnet.
+func (b *BaseSuite) Ghost() *client.GhostClient {
+	ghost := b.Net.ContainerByID(b.GhostID)
+	cli, err := lib.GetGhostClient(ghost)
+	require.NoError(b.T(), err, "could not get ghost client")
+	return cli
+}
+
+// AccessClient returns a client to interact with the access node api on testnet.
+func (b *BaseSuite) AccessClient() *testnet.Client {
+	chain := b.Net.Root().Header.ChainID.Chain()
+	cli, err := testnet.NewClient(fmt.Sprintf(":%s", b.Net.AccessPorts[testnet.AccessNodeAPIPort]), chain)
+	require.NoError(b.T(), err, "could not get access client")
+	return cli
+}
+
+// SetupSuite sets up node configs to run a bare minimum Flow network to function correctly.
+func (b *BaseSuite) SetupSuite() {
```

Comment:
should this generic functionality (to run a bare bones Flow network for testing) be moved to a more central / reusable place (outside `integration/tests/bft`) so that it's re-usable by other integration tests?

Reply:
This test suite is specific to corrupted network, we can probably make a separate generic test suite for other tests. We probably are fine with a shared test suite per package since each test suite in each of the integration /tests packages are slightly differen't.